Tasks/user doc enhancements (#1337)
LLNL/RAJA: docs/sphinx/user_guide/getting_started.rst

Getting Started With RAJA
*************************

This section should help get you up and running with RAJA quickly.

============
Requirements
============

The primary requirement for using RAJA is a C++14 standard compliant compiler.

> Using features, such as various programming model back-ends like CUDA or HIP, 
> requires that they be supported by the compiler you chose. Available RAJA

Reviewer: ```suggestion
Certain features, such as various programming model back-ends like CUDA or HIP, 
must be supported by the compiler you chose. Available RAJA
```

compilation. The compiler will pick the PTX architecture to use in the first 
stage of compilation that is suitable for the SASS architecture you specify.

Alternatively, you may specify the PTX and SASS architectures, using
appropriate nvcc options in the ``CMAKE_CUDA_FLAGS_*`` variables.

.. note:: **RAJA requires a minimum CUDA architecture level of `sm_35` to use
          all supported CUDA features.** Mostly, the architecture level affects
          which RAJA CUDA atomic operations are available and how they are
          implemented inside RAJA. This is described in 
          :ref:`feat-atomics-label`.

          * If you do not specify a value for ``CUDA_ARCH``, it will be set to
            `sm_35` by default and CMake will emit a status message 
            indicating this choice was made.

          * If you give a ``CUDA_ARCH`` value less than `sm_35` (e.g., `sm_30`),
            CMake will report this as an error and stop processing.

Also, RAJA relies on the CUB CUDA utilities library, mentioned earlier, for 
some CUDA back-end functionality. The CUB version included in the CUDA toolkit 
installation is used by default when available. This is the case for CUDA 

> version 11 and later. RAJA includes a CUB submodule that can be used with 
> older versions of CUDA. To use an external CUB installation, provide the 

Reviewer: ```suggestion
version 11 and later. RAJA includes a CUB submodule that is used by default with 
older versions of CUDA. To use an external CUB installation, provide the 
```

Author: fixed

following options to CMake:: 

  cmake \
  ... \
  -DRAJA_ENABLE_EXTERNAL_CUB=On \
  -DCUB_DIR=<path/to/cub> \
  ...

.. note:: The CUDA toolkit version of CUB is
          required for compatibility with the CUDA toolkit version of thrust
          starting with CUDA version 11.0.0. So, if you build
          RAJA with CUDA version 11 or higher, you should use the version of
          CUB contained in the CUDA toolkit version you are using to use 

>           Thrust and to be compatible with libraries that use Thrust.

Reviewer: awkward

Author: fixed
